include/f_core/messaging/c_message_port.h

```diff
@@ -0,0 +1,38 @@
+#ifndef CMESSAGEPORT_H
+#define CMESSAGEPORT_H
+
+#include <zephyr/kernel.h>
+
+template <typename T>
+class CMessagePort {
+public:
+    /**
+     * Constructor
+     */
+    CMessagePort() = default;
+
+    /**
+     * Destructor
+     */
+    virtual ~CMessagePort() = default;
+
+    /**
+     * Send a message
+     * @param message Message to send
+     * @param timeout Time before sending times out
+     * @return Zephyr status code
+     */
+    virtual int Send(const T &message, const k_timeout_t timeout = K_FOREVER) = 0;
+
+    /**
+     * Receive a message
+     * @param message Message to receive
+     * @param timeout Time before receiving times out
+     * @return Zephyr status code
+     */
+    virtual int Receive(T& message, const k_timeout_t timeout = K_FOREVER) = 0;
+};
+
+
```

Comment:
It would be nice to have a wait to receive from multiple message ports. zephyr has the [k_poll_api](https://docs.zephyrproject.org/latest/kernel/services/polling.html) for this and its really useful to say im going to go to sleep until A or B happens without having to sleep, check if anythings available then returning to sleeping. With C++ we can make its API not so clunky to use even. 

We can either implement such a function on the CMsgQueuePort level (pretty straightforward) or at the  CMessagePort level which seems more difficult since we don't know whats gonna be there / how to add it to a poll group.

I was playing around with the API could work for something like that - idk if this is the best option but it is an option. 
https://godbolt.org/z/Y8x6qEWfE

Reply:
Sounds like a good ticket for you to do :)

Reply:
Referencing in #163. Want to just get the basics going first and focus my attention on other sw things.

Reply:
Damn. foiled into doing work

Reply:
I cant be writing all the code 🤠

Reply:
Its Aaron's Software Army, not Aaron's Code Reviewers